Language/Haskell/GhcMod/Monad/Types.hs

```diff
@@ -129,10 +138,11 @@ data GhcModState = GhcModState {
       gmGhcSession   :: !(Maybe GmGhcSession)
     , gmComponents   :: !(Map ChComponentName (GmComponent GMCResolved (Set ModulePath)))
     , gmCompilerMode :: !CompilerMode
+    , gmMMappedFiles :: Map TargetId (StringBuffer, UTCTime)
```

Comment:
Sorry about the hold-up, I caught cold somehow.

On the topic of possible options, after thinking about this for a while, I think `--file-map` should only accept filenames on both left and right, at least for now. This would eliminate trickier corner cases, while being reasonably useful. With this limitation, it becomes somewhat easier.

Guessing if it's filename or module name is relatively cumbersome and error-prone, it could be a better idea in general to add another option for modules altogether.

FYI, on `master` branch of ghc-mod, `ghc-mod check ModuleName` doesn't work: it throws `canonicalizePath: does not exist (No such file or directory)`.